scripts/integration-tests/default.nix

```diff
@@ -0,0 +1,46 @@
+{
+  nixpkgs ? (import ../../.).nixpkgs,
+  binaries ? (import ../../.).binaries,
+  haddock ? (import ../../.).haddock,
+  content ? (import ../../.).content,
+  system ? builtins.currentSystem,
+  lib ? nixpkgs.lib
+}:
+
+let
+  runCuriosity = nixpkgs.writers.writeBashBin "run-curiosity" ''
+    set -euo pipefail
+    PATH=${lib.strings.makeBinPath [ binaries ]}:$PATH
+    STATIC_PATH=${content}
+    ${builtins.readFile ./run-curiosity.sh}
+  '';
+  nginxConf = nixpkgs.substituteAll {
+    src = ./nginx.conf;
+    curiosityaddr = "http://127.0.0.1:9100";
+    # TODO: find a way to reference this folder without the explicit
+    #       0.1.0.0 version
+    curiosityhaddock = "${haddock.doc}/share/doc/curiosity-0.1.0.0/html";
+  };
+  runNGinx = nixpkgs.writers.writeBashBin "run-nginx" ''
+    set -euo pipefail
+    rundir=$(mktemp -d)
+    trap 'rm -rf -- "$rundir"' EXIT
+    cd "$rundir"
+    cp ${nginxConf} nginx.conf
+    mkdir -p nginx-cache
+    sed -i "s~@tmpdir@~$rundir/nginx-cache~g" nginx.conf
+    echo "[+] Starting nginx on port 8888"
```

Comment:
I don't know if it would be actually cleaner, but since you're doing `${builtins.readFile ./run-curiosity.sh}` to craft `runCuriosity` above, and you argue for

> editor syntax color highlighting and linting support.

in your README, I guess I would follow the same practice to craft the `runNginx` script too.